src/Task/Git/CommitMessage.php

```diff
@@ -235,6 +256,27 @@ private function getSpecialPrefixLength($string)
         return mb_strlen($match[0]);
     }
 
+    /**
+     * @param ContextInterface $context
+     *
+     * @return bool
+     */
+    private function subjectHasPunctuations(ContextInterface $context)
+    {
+        $commitMessage = $context->getCommitMessage();
```

Comment:
Probably best to use the `getCommitMessageLinesWithoutComments()` method to fetch the subject. It fixes some issues when the commit message contains comments. Maybe even better to add a private method `getSubjectLine()`, since that functionality is being duplicated in the task.

Reply:
I was simply trying to be consistent with the way the rest of the code was written ...

```php
private function enforceTextWidth(ContextInterface $context)
    {
        $commitMessage = $context->getCommitMessage();
        $config = $this->getConfiguration();

        if (trim($commitMessage) === '') {
            return TaskResult::createPassed($this, $context);
        }

        $errors = [];
        $lines = $this->getCommitMessageLinesWithoutComments($commitMessage);
...
```
Above is only `1 of many` that uses the exact method I used here ?

***Or***

Are you proposing that this be refactored due to the observation of what I am stating above ? I can certainly refactor the whole task to make use of a new private method.

Please advise
Thanks

Reply:
I am sorry, the comment should be one line below on how the subject is subtracted from the message.
The first line can e.g. contain a comment. Since that logic for fetching the subject is repeated, that might go in a new private method. Feel free to refactor where it makes sense.

Reply:
These should resolve this conversation
https://github.com/phpro/grumphp/pull/541/commits/3217efd75d9a477b8f820553add23d6ed8f3a45d
https://github.com/phpro/grumphp/pull/541/commits/b3182857cabc6ef3d075d5c41dbdaf75a05c0483